test/gtest/core/test_mc_reduce.h

```diff
@@ -86,6 +87,31 @@ struct ReductionTest<UCC_DT_FLOAT64, op>
     }
 };
 
+template <template <typename P> class op>
+struct ReductionTest<UCC_DT_BFLOAT16, op> {
+    using type                            = uint16_t;
+    const static ucc_datatype_t     dt    = UCC_DT_BFLOAT16;
+    const static ucc_reduction_op_t redop = op<float>::redop;
+    static void                     assert_equal(type arg1, type arg2)
+    {
+        // near because of different calculation methods - in CPU op across all
+        // vectors as fp32, and only then convert to bfloat16. here conversion to fp32
+        // is per couple. In CUDA, op is as "real" bfloat16 op.
+        // For example - 15*29=435, which doesn't fit in bfloat16.
+        // casting from fp32 will result in 434, while in GPU we will get 436.
+        ASSERT_NEAR(bfloat16tofloat32(&arg1), bfloat16tofloat32(&arg2),
```

Comment:
@lappazos i don't get it: on the one hand you say that there can be a difference up to 2 (434 vs 436) but adding ASSERT_NEAR with the precision e-33. Is 33 decimal digits intended here?

Reply:
So the difference isn't absolute, it depends on the range of values were dealing with - the example I gave is regarding round numbers, where the difference is 2 in that specific case. In the range of values we are using in the test (small floats), this difference out here is enough.